src/app/_components/UserIcon/UserIcon.tsx

```diff
@@ -0,0 +1,172 @@
+    toggleDrawer(true);
+    setAnchorEl(event.currentTarget);
+    setOpen(true);
+  };
+
+  const handleLogOut = () => {
+    onLogOut();
+  };
+
+  const toggleDrawer = (newOpen: boolean) => () => {
+    setOpen(newOpen);
+  };
+
+  const desktopStyling: SxProps = {
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "space-evenly",
+    p: 1.5,
+    borderRadius: 10,
+    "&:hover": {
+      bgcolor: theme.palette.grey[300],
+    },
+    cursor: "pointer",
+  };
+
+  return (
+    <>
+      <ClickAwayListener
+        onClickAway={() => {
+          setOpen(false);
+        }}
+      >
+        <Box
+          sx={
+            isDesktopLargeScreen
+              ? desktopStyling
+              : {
+                  display: "flex",
+                  justifyContent: "center",
+                  alignItems: "center",
+                  cursor: "pointer",
+                }
+          }
+          onClick={handleClick}
+        >
+          <Avatar
+            size={isMobileSmallScreen ? "small" : "medium"}
+            sx={{ mr: 1 }}
+          />
+          {isDesktopLargeScreen && (
+            <>
+              <Box id="name-username" width="65%">
+                <Typography noWrap>{name}</Typography>
+                <Typography
+                  noWrap
+                  variant="body2"
+                  color={theme.palette.grey[500]}
+                >
+                  {username}
+                </Typography>
+              </Box>
+              <MoreHorizIcon fontSize="small" />
+            </>
+          )}
+          {(isDesktopLargeScreen || isTabletMediumScreen) && (
+            <Popper
+              // Note: The following zIndex style is specifically for documentation purposes and may not be necessary in your application.
+              open={open}
+              anchorEl={anchorEl}
+              placement={isDesktopLargeScreen ? "top" : "right"}
+              transition
+            >
+              {({ TransitionProps }) => (
+                <Fade {...TransitionProps} timeout={350}>
+                  <Paper>
+                    <Button
+                      variant="text"
+                      sx={{
+                        p: 2,
+                        cursor: "pointer",
+                        width: "150",
+                        fontWeight: "bold",
+                      }}
+                      onClick={handleLogOut}
+                    >
+                      {`Log Out ${username}`}
+                    </Button>
+                  </Paper>
+                </Fade>
+              )}
+            </Popper>
+          )}
+        </Box>
+      </ClickAwayListener>
+      {/* placed the drawer outsize of the click away listener because it was causing issues with how drawer was closing */}
+      {isMobileSmallScreen && (
+        <Drawer
+          open={open}
+          onClose={toggleDrawer(false)}
+          sx={{ width: "200px", maxWidth: "200px" }}
```

Comment:
Don't need a drawer for mobile, i think we mentioned in the story that mobile will use the same popper that desktop has.